fasthttputil/stringslices.go

```diff
@@ -0,0 +1,107 @@
+package fasthttputil
+
+import (
+	"sync"
+
+	"github.com/valyala/bytebufferpool"
+)
+
+var (
+	stringSlicesPool sync.Pool
+)
+
+// AcquireStringSlices returns an empty StringSlice instance from pool.
+//
+// The returned StringSlice instance may be passed to ReleaseStringSlices
+//  when it is no longer needed.
+func AcquireStringSlices() *StringSlices {
+	v := stringSlicesPool.Get()
+	if v == nil {
+		return &StringSlices{}
+	}
+	return v.(*StringSlices)
+}
+
+// ReleaseStringSlices returns slices acquired via AcquireStringSlices to pool.
+func ReleaseStringSlices(slices *StringSlices) {
+	slices.Reset()
+	stringSlicesPool.Put(slices)
+}
+
+// StringSlices is used to convert byte slices to string slices,
+// it only needs to allocate memory once.
```

Comment:
Can you explain here why it only allocates once and why this is useful?

Reply:
And you should definitely explain here or below that each `WriteBytes` and `NextStringSlice` call must match up as `NextStringSlice` returns things in the same order as `WriteBytes` has written them.

I'm wondering if it would be possible to tag each string. So it would be `s.WriteBytes("method", methodBytes)` and `s.GetString("method")`. This might be hard to do without allocations I guess. It would require `lengths` to be a `map[string]int`.